### PR TITLE
tests: queue modifying and adding test case

### PR DESCRIPTION
**Pull request by Zhaoningx**
1.test_queue_thread2thread(), test_queue_thread2isr(), test_queue_isr2thread()
Test that data passing between ISR and thread by using queue, and add some comments to describe it.
2.test_queue_alloc_prepend_user(), test_queue_alloc_append_user()
Test that a queue's memory can be allocated at running time, and verify the function of "first in,first out" or "first in, last out" by APIs k_queue_alloc_prepend() and k_queue_alloc_append(), and add some comments to describe it.
3.Modify some code styles.
In #ifndef-#endif region at main.c, I think it's a better way to use dummy test to skip some testcases that needn't test in kernel space instead.
4.Add a new testcase test_any_number_queue()
Test that lots of queue can be defined.

Signed-off-by: Ningx Zhao ningx.zhao@intel.com

### CODE REVIEW

File: tests/kernel/queue/src/test_queue_user.c [hunk 1]
```diff
@@ -58,9 +57,11 @@ void child_thread_get(void *p1, void *p2, void *p3)
 }
 
 /**
- * @brief Verify queue elements from a user thread
+ * @brief Verify queue elements and cancle wait from a user thread
```
Comment (andrewboie) [inline tests/kernel/queue/src/test_queue_user.c:60]: "cancel wait"

File: tests/kernel/queue/src/test_queue_user.c [hunk 2]
```diff
@@ -58,9 +57,11 @@ void child_thread_get(void *p1, void *p2, void *p3)
...
  * @details The test adds elements to queue and then
  * verified by the child user thread.
+ * Get data from a empty queue,and use K_FORVER to wait for available
+ * And to cancle wait from another user thread.
```
Comment (andrewboie) [inline tests/kernel/queue/src/test_queue_user.c:64]: "cancel"

File: tests/kernel/queue/src/test_queue_user.c [hunk 3]
```diff
@@ -151,8 +172,10 @@ void test_queue_alloc_append_user(void)
 		zassert_true(qd != NULL, NULL);
 		zassert_equal(qd->data, i, NULL);
 	}
+
```
Comment (andrewboie) [inline tests/kernel/queue/src/test_queue_user.c:175]: can you remove these extra whitespace changes

File: tests/kernel/queue/src/test_queue_user.c (hunk 1 above)
Comment (Zhaoningx) [reply tests/kernel/queue/src/test_queue_user.c:60]: I'm sorry, and I'll be going to pay more attention to spelling, which has been changed.

File: tests/kernel/queue/src/test_queue_user.c (hunk 2 above)
Comment (Zhaoningx) [reply tests/kernel/queue/src/test_queue_user.c:64]: Having been modified, thanks.

File: tests/kernel/queue/src/test_queue_user.c (hunk 3 above)
Comment (Zhaoningx) [reply tests/kernel/queue/src/test_queue_user.c:175]: OK

Comment (Zhaoningx) [reply tests/kernel/queue/src/test_queue_user.c]: hi @andrewboie , I have corrected my mistakes, could you please review it again, thanks.

**Pull request closed without merging**